internal/driver/config.go

```diff
@@ -0,0 +1,90 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+//
+// Copyright (C) 2022 Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package driver
+
+import (
+	"fmt"
+	"reflect"
+	"strconv"
+
+	"github.com/edgexfoundry/go-mod-core-contracts/v2/errors"
+	"github.com/edgexfoundry/go-mod-core-contracts/v2/models"
+)
+
+type configuration struct {
+	CredentialsRetryTime int
+	CredentialsRetryWait int
+	RequestTimeout       int
+	// DefaultAuthMode indicates the Onvif camera default auth mode. "digest" | "usernametoken" | "both" | "none"
+	DefaultAuthMode string
+	// DefaultSecretPath indicates the secret path to retrieve username and password from secret store.
+	DefaultSecretPath string
+	// DiscoveryEthernetInterface indicates the target EthernetInterface for discovering. The default value is `en0`, the user can modify it to meet their requirement.
+	DiscoveryEthernetInterface string
+	// BaseNotificationURL indicates the device service network location
+	BaseNotificationURL string
+}
+
+type CameraInfo struct {
+	Address    string
+	Port       int
+	AuthMode   string
+	SecretPath string
+}
+
+// loadCameraConfig loads the camera configuration
+func loadCameraConfig(configMap map[string]string) (*configuration, error) {
+	config := new(configuration)
+	err := load(configMap, config)
+	if err != nil {
+		return config, err
+	}
+	return config, nil
+}
+
+func CreateCameraInfo(protocols map[string]models.ProtocolProperties) (*CameraInfo, errors.EdgeX) {
```

Comment:
Add godocs style comment to all public func & struct

Reply:
Adds comments